lld/COFF/DLL.cpp

```diff
@@ -395,6 +395,7 @@ class TailMergePDataChunkX64 : public NonSectionChunk {
   }
 
   size_t getSize() const override { return 3 * sizeof(uint32_t); }
+  MachineTypes getMachine() const override { return AMD64; }
```

Comment:
Why is this bit needed here (and the other addition of `getMachine()` as well), or more precisely - why wasn't it needed before and why is it needed now?

Reply:
ARM64EC targets require an additional operation in `Writer::mergeSections` to separate ARM and x86 chunks. Other targets do not consider the machine type of `.pdata` entries, so this was unnecessary prior to this PR.

I added `getMachine` to `TailMergeUnwindInfoX64` for consistency, though it could be skipped.